### PR TITLE
Voting: make inequality strict

### LINKED ISSUE #427: Voting: making inequality for passing votes strict?
Related to https://github.com/aragon/governance/issues/32#issuecomment-416337704
It makes sense for Democracy DAOs but it doesn't for Multisigs, as @sohkai pointed out, e.g., a 2 out of 4 signatures Multisig would require the inequality not to be strict to easily set up a 50%.
An option would be to add a boolean initialization parameter to indicate how inequalities (for both quorum and support) should behave (something like `bool strictThresholds`), which would be set up to by default to `true` for Democracy DAO kit and to `false` for Multisig.
Opening this issue for discussion.

### PR DESCRIPTION
**Pull request by bingen**
Closes #427.

### CODE REVIEW

File: apps/voting/test/voting.js [hunk 1]
```diff
@@ -174,189 +139,234 @@ contract('Voting App', accounts => {
...
+            const holder51 = accounts[2]
+            const nonHolder = accounts[4]
+
+            const neededSupport = pct16(50)
+            const minimumAcceptanceQuorum = pct16(20)
 
             beforeEach(async () => {
+                token = await MiniMeToken.new(NULL_ADDRESS, NULL_ADDRESS, 0, 'n', decimals, 'n', true) // empty parameters minime
+
+                await token.generateTokens(holder20, 20)
```
Comment (izqui) [inline apps/voting/test/voting.js]: Assigned balances should be multiplied by `10 ^ decimals`, otherwise we are just testing the same every time. You can use logic similar to the one in the `pct16` helper

File: apps/voting/test/voting.js [hunk 2]
```diff
@@ -174,189 +139,234 @@ contract('Voting App', accounts => {
...
             return assertRevert(async () => {
                 await voting.changeMinAcceptQuorumPct(neededSupport.plus(1))
             })
         })
 
-        context('creating vote', () => {
-            let script, voteId
+    })
+
+    for (const decimals of [0, 2, 10, 18, 36]) {
```
Comment (izqui) [inline apps/voting/test/voting.js]: Nice one. We probably don't need to test all these cases though, I would test a maximum of 3, otherwise it is going to take for ever to run voting tests

File: apps/voting/test/voting.js [hunk 3]
```diff
@@ -174,189 +139,234 @@ contract('Voting App', accounts => {
...
+                    assert.equal(await voting.getVoteMetadata(voteId), 'metadata', 'should have returned correct metadata')
+                    assert.equal(await voting.getVoterState(voteId, nonHolder), VOTER_STATE.ABSENT, 'nonHolder should not have voted')
+                })
 
-                const state = await voting.getVote(voteId)
-                assert.equal(state[5].toNumber(), neededSupport.toNumber(), 'required support in vote should stay equal')
-                await voting.executeVote(voteId) // exec doesn't fail
-            })
+                it('fails getting a vote out of bounds', async () => {
+                    assertRevert(async () => {
```
Comment (izqui) [inline apps/voting/test/voting.js]: await or return

File: apps/voting/test/voting.js [hunk 4]
```diff
@@ -174,189 +139,234 @@ contract('Voting App', accounts => {
...
+                    })
+                })
 
-            it('vote can be not executed automatically if decided', async () => {
-                await voting.vote(voteId, true, false, { from: holder50 }) // doesnt cause execution
-                await voting.executeVote(voteId)
-                assert.equal(await executionTarget.counter(), 2, 'should have executed result')
-            })
+                it('cannot execute vote if not support met', async () => {
+                    await voting.vote(voteId, false, true, { from: holder29 })
```
Comment (izqui) [inline apps/voting/test/voting.js]: Would be interesting to test when a vote gets 50% support but doesn't pass because of the strict check.

**Comment by bingen** [comment]
Some of them are tested at the end, where it says `isValuePct unit test`. Let me know which ones you want to add and I'll do it.

File: apps/voting/test/voting.js [hunk 5]
```diff
@@ -174,189 +140,234 @@ contract('Voting App', accounts => {
...
                 await voting.changeMinAcceptQuorumPct(neededSupport.plus(1))
             })
         })
 
-        context('creating vote', () => {
-            let script, voteId
+    })
+
+    for (const decimals of [0, 2, 18]) {
+        context('normal token supply, ' + decimals + ' decimals', () => {
```
Comment (sohkai) [inline apps/voting/test/voting.js]: We can use the [backtick syntax](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) here :)
